### PR TITLE
Exa MCP

### PR DESCRIPTION
**Pull request by TimotheeCrouzet**
## Description
Adds an internal MCP server exa with two tools: search_people and search_companies.

search_people: searches Exa's people index (LinkedIn profiles, professional backgrounds) by name, role, or company
search_companies: searches Exa's company index by name, industry, or criteria

## Tests
Tested manually:
- search_people: queried known and unknown people, edge cases (ambiguous names, non-existent people)
- search_companies: queried known companies, ambiguous names, non-existent companies
- Verified the server does not appear in the capabilities picker without the feature flag

## Risk
Low. The server is behind a dust_only feature flag and only available to workspaces that explicitly enable it. Uses a Dust-managed API key with no OAuth. 

## Deploy Plan
none. DUST_MANAGED_EXA_API_KEY is already set in prod

### CODE REVIEW

File: front/lib/actions/mcp_internal_actions/constants.ts [hunk 1]
```diff
@@ -1152,6 +1154,17 @@ export const INTERNAL_MCP_SERVERS = {
     timeoutMs: undefined,
     metadata: WORKSPACE_ANALYTICS_SERVER,
   },
+  exa: {
+    id: 1036,
+    availability: "manual",
+    allowMultipleInstances: false,
+    isPreview: false,
+    isRestricted: ({ featureFlags }) => !featureFlags.includes("exa_mcp"),
```
Comment (spolu) [inline front/lib/actions/mcp_internal_actions/constants.ts:1162]: Since the feature flag pre-exist the PR, does it control anything else?

**Review by spolu — APPROVED** [review]
LGTM

Left a question on the feature flag.
Indentation change looks surprising?

File: front/lib/actions/mcp_internal_actions/constants.ts (hunk 1 above)
Comment (TimotheeCrouzet) [reply front/lib/actions/mcp_internal_actions/constants.ts:1162]: I added the feature flag in this PR so it only controls access to exa MCP:
```
exa_mcp: {
    description: "Access to Exa MCP server (search_people, search_companies)",
    stage: "dust_only",
  },
```

**Comment by TimotheeCrouzet** [comment]
> LGTM
> 
> Left a question on the feature flag. Indentation change looks surprising?

yes indentation is weird, I simply ran `update-internal-mcp-availability-snapshot`. It could be because the previous snapshots updates were made by hand.

Im fixing it so its more compact as expected.

File: front/lib/actions/mcp_internal_actions/constants.ts (hunk 1 above)
Comment (spolu) [reply front/lib/actions/mcp_internal_actions/constants.ts:1162]: ah sorry indeed missed that it was added here 👍

**Comment by spolu** [comment]
Great PR. Good speed! \o/

==> #shipped [INTERNAL] especially to let GTM teams know it exists to try it.

Thinking a bit more about it. Let's rename the MCP server to `exa_people_and_company` (to make it more explicit) given the overlap of "exa" with browse/search

Sorry for the post-merge ask 🙏

**Comment by TimotheeCrouzet** [comment]
> Great PR. Good speed! \o/
> 
> ==> #shipped [INTERNAL] especially to let GTM teams know it exists to try it.
> 
> Thinking a bit more about it. Let's rename the MCP server to `exa_people_and_company` (to make it more explicit) given the overlap of "exa" with browse/search
> 
> Sorry for the post-merge ask 🙏

No problem, I'll change that ! I'll also post a message on #shipped

**Comment by TimotheeCrouzet** [comment]
here is the rename pr: [https://github.com/dust-tt/dust/pull/27262](url)
